src/game/engine/display.cpp

```diff
@@ -1137,15 +1986,127 @@ cell_t DisplayClass::Set_Cursor_Pos(cell_t cell)
     return retval;
 }
 
+/**
+ * Helper function handling the proximity check part.
+ */
+void DisplayClass::Check_Proximity(ObjectTypeClass *object, HousesType house, cell_t cell, int &passes, bool &outside_radar) const
+{
+    BuildingTypeClass *bld = reinterpret_cast<BuildingTypeClass *>(object);
+
+    if (!In_Radar(cell)) {
+        outside_radar = true;
+        passes = 0;
+        return;
+    }
+
+    // check adjacent cells in all directions until we meet the condition to stop
+    for (FacingType i = FACING_NORTH; i < FACING_COUNT; ++i) {
+        cell_t adj_cell = Cell_Get_Adjacent(cell, i);
+        if (In_Radar(adj_cell)) {
+            // is the adjacent cell a Wall or Bib owned by this house?
+            if (bld->Is_Wall() || Array[adj_cell].Has_Bib()) {
+                if (Array[adj_cell].Owner() == house) {
+                    passes = 1;
+                    return;
+                }
+            }
+
+            // is the adjacent cell a building owned by this house and is BaseNormal?
+            BuildingClass *tptr = (BuildingClass *)Array[adj_cell].Cell_Techno(0, 0);
```

Comment:
The arguments for the Cell_Techno() call can be removed as Cell_Techno() has default arguments of the same values.